### PR TITLE
Non-breaking spaces processing

### PR DESCRIPTION
**Pull request by dreikanter**
Hi Aaron, I've fixed a bug with &nbsp; placeholder replacement and added a new option to keep &nbsp; if required (could be helpful for non-google-docs input). Also I've updated the test script to make it work correctly on Windows environment. All tests passed.

### CODE REVIEW

**Comment by aaronsw** [comment]
Can you add a test to make sure this works properly with unicode_snob?

**Comment by dreikanter** [comment]
I've updated the test as you asked (it passed).

Also I've examined unicode_snob parameter more accurately and found out
that my original idea behind keeping &nbsp; in the output is not relevant.
I supposed html2text always converts &nbsp; to ordinary space, but actually
it could be transformed to non-breakable unicode space which is an suitable
solution for me. So I've rolled back the --keep-nbsp parameter. Summarizing
all the above, my patch includes &nbsp_place_holder; fix and as I mentioned
the updated test.

On Thu, Jul 19, 2012 at 5:53 PM, aaronsw <
reply@reply.github.com

File: html2text.py [hunk 1]
```diff
@@ -250,9 +250,8 @@ def close(self):
         self.o('', 0, 'end')
 
         self.outtext = self.outtext.join(self.outtextlist)
-
-        if self.google_doc:
-            self.outtext = self.outtext.replace('&nbsp_place_holder;', ' ');
+        nbsp = unichr(name2cp('nbsp')) if self.unicode_snob else u' '
```
Comment (aaronsw) [inline html2text.py]: Would you mind converting this to a normal if/else block so we can preserve our backward-compatibility with old Python versions?

Comment (dreikanter) [inline html2text.py]: Done.

On Fri, Jul 20, 2012 at 6:38 PM, aaronsw <
reply@reply.github.com

**Comment by aaronsw** [comment]
Thanks so much for the patch and putting up with my nitpicking!

**Comment by dreikanter** [comment]
You're welcome :)

On Fri, Jul 20, 2012 at 6:52 PM, aaronsw <
reply@reply.github.com

**Pull request merged**